components/omega/doc/design/VertAdv.md

```diff
@@ -0,0 +1,343 @@
+(omega-design-vert-adv)=
+# Vertical Advection
+
+## 1 Overview
+
+The Vertical Advection module is responsible for the vertical transport of mass,
+momentum, and tracers within Omega. This implementation is consistent with
+Omega's non-Boussinesq governing equations, where the prognostic vertical
+coordinate is a pseudo-height, effectively a normalized pressure coordinate. The
+methods compute the tendencies associated with the vertical advection terms in
+the governing equations, and are designed to allow for flexibility in numerical
+scheme configuration.
+
+## 2 Requirements
+
+### 2.1 Requirement: Compute transport pseudo-velocity
+
+Compute the transport pseudo-velocity through vertical layer interfaces based on
+the divergence of the horizontal velocity field. Allow for possible movement of
+layer interfaces and tilt of layer interfaces
+
+### 2.2 Requirement: Advection scheme flexibility
+
+Support for multiple possible reconstruction schemes, allowing users to select
+among different orders of accuracy and upwinding at runtime through
+configuration options. Standard and monotonic flux-corrected transport (FCT)
+advection algorithms are available for the tendency calculations.
+
+### 2.3 Requirement: Compute tendencies for mass, momentum, and tracers
+
+Separate methods are neededto compute the vertical-advection tendencies
+for mass, momentum, and tracers. Mass (`LayerThickness`) and horizontal
+momentum (`NormalVelocity`) are stored as 2-D arrays. `LayerThickness` is
+cell-centered, while `NormalVelocity` is defined on edges. These tendencies
+use a first-order flux formulation that ensures strict conservation and energy
```

Comment:
do we really use first order?  I thought thickness advection was centered (2nd order), but I could be misremembering